src/libraries/System.Private.CoreLib/src/System/RuntimeType.cs

```diff
@@ -139,8 +139,28 @@ public override Array GetEnumValues()
             return ret;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public override Type GetEnumUnderlyingType()
         {
+            if (RuntimeHelpers.IsKnownConstant(this))
+            {
+                switch (Type.GetTypeCode(this))
+                {
+                    case TypeCode.Boolean: return typeof(bool);
+                    case TypeCode.Char:    return typeof(char);
```

Comment:
These two are possible in IL but not in C# (`CS1008`), should we also have nint / nuint, since they are possible in IL?

```C#
.class private auto ansi sealed Foo
    extends [System.Runtime]System.Enum
{
    .field public specialname rtspecialname [System.Runtime]System.UIntPtr value__
    .field public static literal valuetype Foo Bar = uint64(0)
}
```

Reply:
There's no TypeCode for nint/nuint, those have to go through the fallback (in earlier iterations of the code I've noticed we do actually have nint enum tests when my code errored out on them).